### PR TITLE
Fix generate codemod function input and output

### PR DESCRIPTION
**Pull request by ppashakhanloo**
**Please check if the PR fulfills these requirements**:

-   [ x ] The commit message follows [our code of conduct](https://github.com/codemod-com/codemod-registry/blob/main/CODE_OF_CONDUCT.md) and [contributing guidelines](https://github.com/codemod-com/codemod-registry/blob/main/CONTRIBUTING.md)
-   [ ] Tests for the changes have been added (for bug fixes/features)
-   [ x ] Docs have been added / updated (for bug fixes / features)

**Please include the following information in your pull request**:

-   **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This PR changes the input and output of the main function that generates codemods.
I also did some refactoring and changed the default options.

-   **What is the current behavior?** (You can also link to an open issue here)

-   **What is the new behavior (if this is a feature change)?**

-   **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

-   **Any other information (if needed)**:

### CODE REVIEW

File: apps/ai/app.py [hunk 1]
```diff
@@ -40,13 +37,11 @@ async def websocket_status(websocket: WebSocket):
...
             before = data.get('input')
             after = data.get('after')
-            create_ts_file('tests/before_tmp.ts', before)
-            create_ts_file('tests/after_tmp.ts', after)
             async def logger(message):
                 print(message)
                 await websocket.send_json(message)
                 await asyncio.sleep(0)
-            result = await generate_codemod('tests/before_tmp.ts', 'tests/after_tmp.ts', '.temp', configs.cmd_args.max_correction_attempts, configs.cmd_args.llm_engine, configs.cmd_args.codemod_engine, logger)
+            result = await generate_codemod(before, after, gen_uid(), configs.cmd_args.max_correction_attempts, configs.cmd_args.llm_engine, configs.cmd_args.codemod_engine, logger)
```
[the remark this answers is not in the document]
Comment (ppashakhanloo) [reply apps/ai/app.py:44]: Yes. Max correction attempts can be from 0 to 5.

File: apps/ai/iterative_ai.py [hunk 2]
```diff
@@ -52,38 +47,39 @@ async def generate_codemod(before_path, after_path, codemod_path, max_correction
...
+    codemod_path = f'codemod-{uid}.ts'
+    codemod = generate_draft_codemod(before_source, after_source, codemod_path, prompts_thread, llm_engine, codemod_engine)
     common_utils.save_source(codemod, path=codemod_path)
 
-    actual_after_path = '.actual_after.ts'
     if max_correction_attempts == 0:
         await logger({"execution_status": "finished",
             "message": "Draft codemod generated."})
-        return test_generated_codemod(before_path, after_path, codemod_path, actual_after_path, codemod_engine)
+        common_utils.remove(codemod_path)
```
[the remark this answers is not in the document]
Comment (ppashakhanloo) [reply apps/ai/iterative_ai.py:57]: I could not find a way to pass the codemod as a string to jscodeshift and tsc. I think we have to use files here.

File: apps/ai/configs.py [hunk 3]
```diff
@@ -1,43 +1,37 @@
...
 
-  # Codemod Engine Configurations
-  if cmd_args.codemod_engine == 'jscodeshift':
-    import prompt_utils.jscodeshift
-    engine_utils = prompt_utils.jscodeshift
-  else:
-    print("Unsupported codemod engine:", cmd_args.codemod_engine)
-    sys.exit()
+def get_codemod_engine(codemod_engine):
+  if codemod_engine == 'jscodeshift':
```
[the remark this answers is not in the document]
Comment (ppashakhanloo) [reply apps/ai/configs.py:34]: I don't understand what kind of change you are suggesting here. Can you please explain it a bit more?

File: apps/ai/iterative_ai.py [hunk 4]
```diff
@@ -52,38 +47,39 @@ async def generate_codemod(before_path, after_path, codemod_path, max_correction
     await logger({"execution_status": "in-progress",
         "message": "Generating a draft codemod......"})
 
-    codemod = generate_draft_codemod(before, after, codemod_path, prompts_thread, llm_engine, codemod_engine)
+    codemod_path = f'codemod-{uid}.ts'
+    codemod = generate_draft_codemod(before_source, after_source, codemod_path, prompts_thread, llm_engine, codemod_engine)
```
[the remark this answers is not in the document]
Comment (ppashakhanloo) [reply apps/ai/iterative_ai.py:51]: I think it is better to keep this variable's name `codemod`. Later on in the for-loop, we are accessing it again. So we would have to assign `draft_codemod` to `codemod` before entering the loop which is a bit confusing/messy.

File: apps/ai/configs.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (grzim) [reply apps/ai/configs.py:35]: this is about making it more flexible when we want to introduce more engines in the future, but we can wait with this change when we actually add them.

File: apps/ai/iterative_ai.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (grzim) [reply apps/ai/iterative_ai.py:56]: I will take a look on this but sure, let keep it this way for now

**Review by grzim — APPROVED** [review]

**Pull request merged**